providers/env-var/src/main/java/dev/openfeature/contrib/providers/envvar/EnvironmentGateway.java

```diff
@@ -0,0 +1,11 @@
+package dev.openfeature.contrib.providers.envvar;
+
+/**
+ * This is an abstraction to fetch an environment variables. It can be used to support
+ * environment-specific access or provide additional functionality, like prefixes,
+ * casing and even sources like spring configurations which come from different sources.
+ * Also, a test double could implement this interface, making the tests independent of the actual environment.
+ */
+public interface EnvironmentGateway {
```

Comment:
We can mark this as a `@FunctionalInterface` and make the contract even more strict :)